src/libfetchers/tarball.cc

```diff
@@ -220,8 +225,12 @@ struct CurlInputScheme : InputScheme
             if (auto n = string2Int<uint64_t>(*i))
                 input.attrs.insert_or_assign("revCount", *n);
 
-        url.query.erase("rev");
-        url.query.erase("revCount");
+        if (auto i = get(url.query, "lastModified"))
+            if (auto n = string2Int<uint64_t>(*i))
+                input.attrs.insert_or_assign("lastModified", *n);
+
+        for (auto & param : allowedAttrs())
+            url.query.erase(param);
```

Comment:
```suggestion
        /* The URL query parameters serve two roles: specifying fetch settings for Nix itself, and arbitrary data as part of the HTTP request. Now that we've processed the Nix-specific attributes above, remove them so we don't also send them as part of the HTTP request. */
        for (auto & param : allowedAttrs())
            url.query.erase(param);
```
(probably want to wrap lines, maybe move some of this above to where the Nix-specific ones are parsed.)